src/web/app/pages-instructor/instructor-session-individual-extension-page/individual-extension-confirm-modal/individual-extension-confirm-modal.component.ts

```diff
@@ -0,0 +1,143 @@
+  SortOrder: typeof SortOrder = SortOrder;
+  sortStudentsBy: SortBy = SortBy.SESSION_END_DATE;
+  sortStudentOrder: SortOrder = SortOrder.DESC;
+  sortInstructorsBy: SortBy = SortBy.SESSION_END_DATE;
+  sortInstructorOrder: SortOrder = SortOrder.DESC;
+
+  DATETIME_FORMAT: string = 'd MMM YYYY h:mm:ss';
+
+  isNotifyIndividuals: boolean = false;
+
+  onExtend(): void {
+    this.onConfirmExtensionCallBack.emit(this.isNotifyIndividuals);
+  }
+
+  onDelete(): void {
+    this.onConfirmExtensionCallBack.emit(this.isNotifyIndividuals);
+  }
+
+  isDeleteModal() : boolean {
+    return this.modalType === ExtensionModalType.DELETE;
+  }
+
+  isExtendModal() : boolean {
+    return this.modalType === ExtensionModalType.EXTEND;
+  }
+
+  sortStudentColumnsBy(by: SortBy): void {
+    this.sortStudentsBy = by;
+    this.sortStudentOrder = this.sortStudentOrder === SortOrder.DESC ? SortOrder.ASC : SortOrder.DESC;
+    this.selectedStudents.sort(this.sortStudentPanelsBy(by));
+  }
+
+  sortStudentPanelsBy(by: SortBy): (a: StudentExtensionTableColumnModel, b: StudentExtensionTableColumnModel)
+    => number {
+    return (a: StudentExtensionTableColumnModel, b: StudentExtensionTableColumnModel): number => {
+      let strA: string;
+      let strB: string;
+      switch (by) {
+        case SortBy.SECTION_NAME:
+          strA = a.sectionName;
+          strB = b.sectionName;
+          break;
+        case SortBy.TEAM_NAME:
+          strA = a.sectionName;
+          strB = b.sectionName;
+          break;
+        case SortBy.RESPONDENT_NAME:
+          strA = a.name;
+          strB = b.name;
+          break;
+        case SortBy.RESPONDENT_EMAIL:
+          strA = a.email;
+          strB = b.email;
+          break;
+        case SortBy.SESSION_END_DATE:
+          strA = a.extensionDeadline.toString();
+          strB = b.extensionDeadline.toString();
+          break;
+        default:
+          strA = '';
+          strB = '';
+      }
+      return this.tableComparatorService.compare(by, this.sortStudentOrder, strA, strB);
+    };
+  }
+
+  sortInstructorsColumnsBy(by: SortBy): void {
+    this.sortInstructorsBy = by;
+    this.sortInstructorOrder = this.sortInstructorOrder === SortOrder.DESC ? SortOrder.ASC : SortOrder.DESC;
+    this.selectedInstructors.sort(this.sortInstructorPanelsBy(by));
+  }
+
+  sortInstructorPanelsBy(by: SortBy): (a: InstructorExtensionTableColumnModel, b: InstructorExtensionTableColumnModel)
+  => number {
+    return (a: InstructorExtensionTableColumnModel, b: InstructorExtensionTableColumnModel): number => {
+      let strA: string;
+      let strB: string;
+      switch (by) {
+        case SortBy.RESPONDENT_NAME:
+          strA = a.name;
+          strB = b.name;
+          break;
+        case SortBy.RESPONDENT_EMAIL:
+          strA = a.email;
+          strB = b.email;
+          break;
+        case SortBy.INSTRUCTOR_PERMISSION_ROLE:
+          strA = a.role ? a.role.valueOf() : '';
```

Comment:
To check: `valueOf()` may not be necessary. The corresponding method in `InstructorSessionIndividualExtensionPageComponent` didn't have this.